app/transformations/ETMPToFrontEnd/API1832.scala

```diff
@@ -64,147 +63,109 @@ object API1832 {
-        pathUaLumpSumAmountNested.json.copyFrom(pathEtmpAmountLumpSum.json.pick) and
-        pathUaLumpSumDateNested.json.copyFrom(pathEtmpEventDate.json.pick)
+  lazy val readsEvent8PaymentDetails: Reads[JsObject] = (
+      reqNestedReadsJsString(pathUaTypeOfProtection, readsTypeOfProtectionEvent8) and
+      reqReads(pathUaTypeOfProtectionReference, pathEtmpFreeText) and
+      reqReads(pathUaLumpSumAmountNested, pathEtmpAmountLumpSum) and
+      reqReads(pathUaLumpSumDateNested, pathEtmpEventDate)
       ).reduce
-  }
-
-  def yesNoTransform(js: JsValue, err: String):Reads[JsBoolean] = {
-    js match {
-      case JsString(value) => value match {
-        case "Yes" => Reads.pure(JsBoolean(true))
-        case "No" => Reads.pure(JsBoolean(false))
-        case _ => Reads.failed(err)
-      }
-      case _ => Reads.failed(err)
-    }
-  }
 
-  lazy val readsEvent24PaymentDetails: Reads[JsObject] = {
-    (
-      pathUaMemberHoldProtection.json.copyFrom(pathEtmpMemberHoldProtectionEvent24.json.pick.flatMap(yesNoTransform(_, "hold protection failed"))) and
-        pathUaTypeOfProtectionEvent24.json.copyFrom(readsTypeOfProtectionEvent24).orElse(doNothing) and
-        pathUaProtectionGroup1Event24.json.copyFrom(readsTypeOfProtectionGroup1Event24).orElse(doNothing) and
-        pathUaPreCommenceReference.json.copyFrom(readsPreCommenceReference).orElse(doNothing) and
-        pathUaPensionCreditReference.json.copyFrom(readsPensionCreditReference).orElse(doNothing) and
-        pathUaNonResidenceReference.json.copyFrom(readsNonResidenceReference).orElse(doNothing) and
-        pathUaOverseasReference.json.copyFrom(readsOverseasReference).orElse(doNothing) and
-        pathUaAvailableLumpSumDBAExceeded.json.copyFrom(pathEtmpAvailableLumpSumDBAExceeded.json.pick.flatMap(yesNoTransform(_, "available lump sum DBA exceeded failed"))) and
-        pathUaAvailableLumpSumExceeded.json.copyFrom(pathEtmpAvailableLumpSumExceeded.json.pick.flatMap(yesNoTransform(_, "available lump sum exceeded failed"))).orElse(doNothing) and
-        pathUaAmountCrystalised.json.copyFrom(pathEtmpAmountCrystalised.json.pick) and
-        pathUaBCEType.json.copyFrom(readsBCETypeEvent24) and
-        pathUaCrystallisedDateEvent24.json.copyFrom(pathEtmpTaxYearEndingDate.json.pick) and
-        pathUaFreeTextEvent24.json.copyFrom(pathEtmpFreeText.json.pick).orElse(doNothing) and
-        pathUaMarginalRate.json.copyFrom(pathEtmpMarginalRate.json.pick.flatMap(yesNoTransform(_, "taxed at marginal rate failed"))).orElse(doNothing) and
-        pathUaPayeReference.json.copyFrom(pathEtmpPayeReference.json.pick).orElse(doNothing)
-    ).reduce
-  }
+  lazy val readsEvent24PaymentDetails: Reads[JsObject] = (
+    reqReadsBoolTransform(pathUaMemberHoldProtection, pathEtmpMemberHoldProtectionEvent24, yesNoTransformToBoolean) and
+      optNestedReadsJsString(pathUaTypeOfProtectionEvent24, readsTypeOfProtectionEvent24) and
```

Comment:
The yesNoTransformToBoolean is not equivalent. Can we re check other instances of yesNoTransformToBoolean use-case.

Reply:
We weren't passing it into  `reqReadsBoolTransform` before so it needed to give a JsBoolean result rather than just a boolean. We had a lot of different methods transforming booleans so was trying to clean them up a bit.

Reply:
I am okay with this as long as we have unit tests that test that this is not broken.

Reply:
I think this would fail if it was broken - [https://github.com/hmrc/pension-scheme-event-reporting/blob/7f3a11a9df9594770d7938dbf6c5ab79fef9ffbb/test/utils/GeneratorAPI1832.scala#L490]( https://github.com/hmrc/pension-scheme-event-reporting/blob/7f3a11a9df9594770d7938dbf6c5ab79fef9ffbb/test/utils/GeneratorAPI1832.scala#L490)